examples/knowledge-base/components/SearchBox/SearchBox.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState } from "react";
+import { useRouter } from 'next/router';
+import { Autocomplete, Text, Flex } from "@contentful/f36-components";
+import lunr from 'lunr';
+import { css } from "emotion";
+import { ResultType } from "./types";
+
+const pages: ResultType[] = [{
+  url: 'http://localhost:3000',
+  title: 'Title',
+  content: 'Build and extend Contentful products.'
+},{
+  url: 'http://localhost:3000#1',
+  title: 'Title 2',
+  content: 'Some content to show up for title 2'
+},{
+  url: 'http://localhost:3000#2',
+  title: 'Title 3',
+  content: 'Another different content for title 3 page'
+}];
+
+const styles = {
+  searchResults: css({
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    whiteSpace: "pre",
+  }),
+};
+
+const truncateContent = (found, page) => {
+  if(!found || !page) return page;
+  const key = Object.keys(found?.matchData?.metadata).find(key => {
+    console.log(found.matchData.metadata[key])
+    return found.matchData.metadata[key].content?.position
+  });
+  if(!key) return page;
+  const [index] = found.matchData.metadata[key].content.position[0];
+  const startIndex = Math.max(0, index - 15)
+  const truncatedContent = page.content.substring(startIndex, 50);
+  return {
+    ...page,
+    content: startIndex > 0 ? `…${truncatedContent}` : truncatedContent,
+  }
+}
+
+export const SeachBox = () => {
+  const [results, setResults] = useState<ResultType[]>([]);
+  const router = useRouter();
+
+  const idx = lunr(function(builder) {
+    builder.ref('url');
+    builder.field('title');
+    builder.field('content');
+    builder.metadataWhitelist = ['position']
+
+    pages.forEach(page => {
+      builder.add(page);
+    })
+  });
+
+  const handleInputValueChange = (value) => {
```

Comment:
I'm getting matches only after the sixth letter 🤔 Is there a way to improve this?

<img width="809" alt="Screenshot 2022-06-21 at 17 59 02" src="https://user-images.githubusercontent.com/6163988/174844880-1790f3fb-2bbf-42fe-8d5c-097faf96b410.png">
<img width="750" alt="Screenshot 2022-06-21 at 18 00 07" src="https://user-images.githubusercontent.com/6163988/174845093-0a8a8334-6598-46c3-acc2-1aa0a8ce46f9.png">